### ISSUE TITLE
SPL compile failure with 3.6.18 kernel

### ISSUE THREAD

**Issue by wallrj** [zfsonlinux/spl#434]
The compilation of the SPL module fails against a 3.6.18 kernel.

```
tail -n20 /var/lib/dkms/spl/0.6.3/build/make.log 
make[2]: Entering directory `/var/lib/dkms/spl/0.6.3/build/module'
make -C /lib/modules/3.18.7-100.fc20.x86_64/build SUBDIRS=`pwd`  CONFIG_SPL=m modules
make[3]: Entering directory `/usr/src/kernels/3.18.7-100.fc20.x86_64'
  CC [M]  /var/lib/dkms/spl/0.6.3/build/module/spl/../../module/spl/spl-debug.o
  CC [M]  /var/lib/dkms/spl/0.6.3/build/module/spl/../../module/spl/spl-proc.o
  CC [M]  /var/lib/dkms/spl/0.6.3/build/module/spl/../../module/spl/spl-kmem.o
/var/lib/dkms/spl/0.6.3/build/module/spl/../../module/spl/spl-kmem.c: In function ‘spl_kmem_cache_reap_now’:
/var/lib/dkms/spl/0.6.3/build/module/spl/../../module/spl/spl-kmem.c:2368:2: error: implicit declaration of function ‘smp_mb__after_clear_bit’ [-Werror=implicit-function-declaration]
  smp_mb__after_clear_bit();
  ^
cc1: some warnings being treated as errors
make[5]: *** [/var/lib/dkms/spl/0.6.3/build/module/spl/../../module/spl/spl-kmem.o] Error 1
make[4]: *** [/var/lib/dkms/spl/0.6.3/build/module/spl] Error 2
make[3]: *** [_module_/var/lib/dkms/spl/0.6.3/build/module] Error 2
make[3]: Leaving directory `/usr/src/kernels/3.18.7-100.fc20.x86_64'
make[2]: *** [modules] Error 2
make[2]: Leaving directory `/var/lib/dkms/spl/0.6.3/build/module'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/var/lib/dkms/spl/0.6.3/build'
make: *** [all] Error 2
```

This change looks related: https://github.com/zfsonlinux/spl/commit/a988a35a93671c086c38ce5b71b6badb59a9c2de

```
# yum info kernel kernel-devel

Name        : kernel
Arch        : x86_64
Version     : 3.18.7
Release     : 100.fc20
Size        : 141 M
Repo        : installed
From repo   : updates
Summary     : The Linux kernel
URL         : http://www.kernel.org/
Licence     : GPLv2 and Redistributable, no modification permitted
Description : The kernel package contains the Linux kernel (vmlinuz), the core of any
            : Linux operating system.  The kernel handles the basic functions
            : of the operating system: memory allocation, process allocation, device
            : input and output, etc.

Name        : kernel-devel
Arch        : x86_64
Version     : 3.18.7
Release     : 100.fc20
Size        : 34 M
Repo        : installed
From repo   : /kernel-devel-3.18.7-100.fc20.x86_64
Summary     : Development package for building kernel modules to match the kernel
URL         : http://www.kernel.org/
Licence     : GPLv2 and Redistributable, no modification permitted
Description : This package provides kernel headers and makefiles sufficient to build modules
            : against the kernel package.

```

**Comment by kernelOfTruth** [comment]
like mentioned in #441 please give the KMOD approach a try: http://zfsonlinux.org/generic-deb.html

**Comment by ryao** [comment]
This should have been fixed by a988a35a93671c086c38ce5b71b6badb59a9c2de. We are building against a snapshot of HEAD from November internally, which is why @wallrj encountered it.

**Comment by behlendorf** [comment]
Closing.  Should be resolve by e3020723dc43af2bc22af0d68571a61daf9b44d0.
